### ISSUE TITLE
Cross entropy and softmax with an optimizer causes NANs

### ISSUE THREAD

**Issue by ssitu** [ssitu/Neural-Network-Library-Class#3]
A network with a softmax activation function in the output layer with the cross-entropy loss function tends to cause the weights to become NANs when using an adaptive learning rate optimizer with the exception of RMSProp. Not sure if it is because of the way I coded something or if that is its natural behavior.
